cpp/unreal_plugins/SpEngine/Source/SpEngine/GameWorldService.cpp

```diff
@@ -0,0 +1,49 @@
+//
```

Comment:
Since we're grouping service functionality so cleanly and logically now, less documentation is needed.

```
//
// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
//

#include "SpEngine/GameWorldService.h"

#include <string>
#include <vector>

#include <Delegates/IDelegateInstance.h> // FDelegateHandle
#include <Engine/Engine.h>               // GEngine
#include <Engine/World.h>                // UWorld

#include "SpCore/Assert.h"
#include "SpCore/Log.h"
#include "SpEngine/EngineService.h"

void GameWorldService::postWorldInitializationHandler(UWorld* world, const UWorld::InitializationValues initialization_values)
{
    SP_LOG_CURRENT_FUNCTION();
    SP_ASSERT(world);

    #if WITH_EDITOR // defined in an auto-generated header
        bool is_game_world = world->IsGameWorld();
    #else
        bool is_game_world = GEngine->GetWorldContextFromWorld(world) != nullptr;
    #endif

    if (is_game_world) {
        SP_ASSERT(!world_);
        world_ = world;
    }
}

void GameWorldService::worldCleanupHandler(UWorld* world, bool session_ended, bool cleanup_resources)
{
    SP_LOG_CURRENT_FUNCTION();
    SP_ASSERT(world);

    if (world == world_) {
        world_ = nullptr;
    }
}
```